guides/micronaut-cloud-secrets-oracle/micronaut-cloud-secrets-oracle.adoc

```diff
@@ -0,0 +1,457 @@
+
+In this guide we'll use Vault to store database connection information for a Micronaut application that uses MySQL, and use the Micronaut support for Oracle Cloud to make the process seamless.
+
+== The Application
+
+Download the complete solution of the guideLink:micronaut-data-jdbc-repository[Access a database with Micronaut Data JDBC] guide. You will use the sample application as a starting point.
+
+== Creating an Oracle Cloud Vault
+
+We'll do the work of creating the vault with the Oracle Cloud CLI. See the https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.7.1/oci_cli_docs/cmdref/kms.html[Vault CLI command reference] for more information.
+
+NOTE: If you prefer to use the Oracle Cloud web console to create the vault, follow the steps in the "Using Oracle Cloud Vault" section of the guideLink:micronaut-oracle-autonomous-db[Access an Oracle Autonomous Database] guide.
+
+There are three required parameters for the create command; compartment OCID, display name, and vault type.
+
+common:oracle-cloud-compartment-id.adoc[]
+
+=== Create the Vault
+
+Choose a display name (1-100 characters) and a vault type, either `DEFAULT` or `VIRTUAL_PRIVATE`. We recommend that you choose `DEFAULT` to avoid the cost of a virtual private vault.
+
+Run the https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.7.1/oci_cli_docs/cmdref/kms/management/vault/create.html[create command] with the compartment OCID, display name, and vault type substituted:
+
+[source,bash]
+----
+oci kms management vault create -c $C \
+    --display-name mn_guide_vault \
+    --vault-type DEFAULT
+----
+
+The response should look like this:
+
+[source,json]
+----
+{
+  "data": {
+    "compartment-id": "ocid1.tenancy.oc1..aaaaaaaaixksuiqo3rx6...",
+    "crypto-endpoint": null,
+    "defined-tags": {},
+    "display-name": "mn_guide_vault",
+    "freeform-tags": {},
+    "id": "ocid1.vault.oc1.iad.b5re3....abuwcljryp5dmxbu3si7...",
+    "is-primary": true,
+    "lifecycle-state": "CREATING",
+    "management-endpoint": null,
+    "replica-details": null,
+    "restored-from-vault-id": null,
+    "time-created": "2022-04-05T12:35:22.665000+00:00",
+    "time-of-deletion": null,
+    "vault-type": "DEFAULT",
+    "wrappingkey-id": ""
+  },
+  "etag": "b577fd72830444583aa6..."
+}
+----
+
+Save the vault OCID from the `id` property in the response; this is the unique identifier for your vault and you'll need it later.
+
+==== Create a Master Encryption Key
+
+Next we need to https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Tasks/managingkeys.htm[create a Master Encryption Key].
+
+For this we need the management endpoint URL that wasn't available yet in the response from the vault `create` command. Run the https://docs.oracle.com/en-us/iaas/tools/oci-cli/2.9.4/oci_cli_docs/cmdref/kms/management/vault/get.html[get command] with the vault OCID substituted:
+
+[source,bash]
+----
+oci kms management vault get --vault-id ocid1.vault.oc1.iad.b5re3....abuwcljryp5dmxbu3si7...
```

Comment:
Same here.

Reply:
I left this one since they're also looking for the `lifecycle-state`

Reply:
In the AWS guide, I included this command that is natively supported by the AWS CLI:

```bash
aws rds wait db-instance-available --db-instance-identifier micronaut-guides-mysql
```

Assuming that nothing similar exists for the OCI CLI, you could achieve something similar with:

```bash
until [ $(oci kms management vault get --vault-id $VAULT_ID | jq -r '.data.lifecycle-state') = *"ACTIVE"* ] ; do echo 
 "Waiting for Vault to be active" && sleep 5 ; done
```